### PR TITLE
Added Icon directive

### PR DESCRIPTION
**Pull request by peek4y**
Added a new directive for icons - <code>sm-icon</code>

**Attributes**
 - <code>sm-font-icon</code> : specify the font icon class.
 - <code>sm-icon-size</code> : size of the icon. (in px / em)

### CODE REVIEW

**Comment by peek4y** [comment]
Please review.

**Comment by peek4y** [comment]
@m0t0r  Please review.

File: src/elements/button/button.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 'use strict';
 
 angular
-  .module('semantic.ui.elements.button', [])
+  .module('semantic.ui.elements.button', ['semantic.ui.elements.icon', 'semantic.ui.elements.divider'])
```
Comment (m0t0r) [inline src/elements/button/button.js]: Changes that you made in _**'semantic.ui.elements.button'**_ module are not related to your initial PR and _should not_ be here

File: src/elements/icon/demo/icons.html [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<!DOCTYPE html>
+<html ng-app="semantic.ui.angular.icons">
+<head lang="en">
+    <meta charset="UTF-8">
+    <title>Semantic-UI Demo page</title>
+    <link rel="stylesheet" href="../../../../docs/config/assets/semantic-ui/semantic.css"/>
+</head>
+<body style="padding:100px;text-align:center;">
+
+<sm-icon sm-font-icon="child icon" ng-style="{ color: 'red' }"></sm-icon>
```
Comment (m0t0r) [inline src/elements/icon/demo/icons.html]: I do not see much of the meaning to have a directive if you have to set even base css class - _**icon**_

File: src/elements/button/button.js (hunk 1 above)
Comment (peek4y) [inline src/elements/button/button.js]: Oh yes! Sorry about that, i'll move those dependencies to demo/button.module.js

Comment (peek4y) [inline src/elements/button/button.js]: Or I'm not quite not sure by what "should not be here" means, you want me to create a new PR ?

File: src/elements/icon/demo/icons.html (hunk 2 above)
Comment (peek4y) [inline src/elements/icon/demo/icons.html]: I was wondering if we could add more features later like svg Icons, hence added the directive.

Comment (peek4y) [inline src/elements/icon/demo/icons.html]: Right got it.

File: src/elements/button/button.js (hunk 1 above)
Comment (m0t0r) [inline src/elements/button/button.js]: your PR is about a new directive - sm-icon. You made changes in another module and are not related to the new feature, if you want to propose some changes for another element (module) it should be in a separated PR.

Comment (peek4y) [inline src/elements/button/button.js]: Sorry, I'm a bit new to PRs, when I tried creating a new PR, for the buttons module, it got merged with the current one :(

Comment (caitp) [inline src/elements/button/button.js]: It's a safe bet to always create a new branch for any feature or fix you're working on, no matter how small

Comment (peek4y) [inline src/elements/button/button.js]: @caitp Got it!

**Comment by m0t0r** [comment]
To me, this is not ready to be merged. @peek4y I mentioned before that we should do changes in a unrelated modules in a separate PRs. I still see changes on the button directive when your PR is about Icon directive

**Comment by peek4y** [comment]
@m0t0r  How can I remove those unwanted commits?

**Comment by peek4y** [comment]
I'll start off fresh.

**Pull request closed without merging**